tachyon/zk/plonk/prover/argument.h

```diff
@@ -0,0 +1,189 @@
+          absl::MakeConstSpan(advice_columns_vec_[i]);
+      absl::Span<const Evals> instance_columns =
+          absl::MakeConstSpan(instance_columns_vec_[i]);
+      Table<Evals> circuit_table(fixed_columns, advice_columns,
+                                 instance_columns);
+      ret.push_back(std::move(circuit_table));
+    }
+    return ret;
+  }
+
+  // Return a table including every type of polynomials in coefficient form.
+  std::vector<Table<Poly>> ExportPolyTables() const {
+    CHECK(advice_switched_);
+    std::vector<Table<Poly>> ret;
+    ret.reserve(num_circuits_);
+
+    absl::Span<const Poly> fixed_polys = absl::MakeConstSpan(*fixed_polys_);
+    for (size_t i = 0; i < num_circuits_; ++i) {
+      absl::Span<const Poly> advice_polys =
+          absl::MakeConstSpan(advice_polys_vec_[i]);
+      absl::Span<const Poly> instance_polys =
+          absl::MakeConstSpan(instance_polys_vec_[i]);
+      Table<Poly> circuit_table(fixed_polys, advice_polys, instance_polys);
+      ret.push_back(std::move(circuit_table));
+    }
+    return ret;
+  }
+
+  const std::vector<F>& GetAdviceBlinds(size_t circuit_idx) const {
+    CHECK_LT(circuit_idx, num_circuits_);
+    return advice_blinds_vec_[circuit_idx];
+  }
+
+  const std::vector<F>& challenges() const { return challenges_; }
+
+ private:
+  Argument(size_t num_circuits, const std::vector<Evals>* fixed_columns,
+           const std::vector<Poly>* fixed_polys,
+           std::vector<std::vector<Evals>>&& advice_columns_vec,
+           std::vector<std::vector<F>>&& advice_blinds_vec,
+           std::vector<std::vector<Evals>>&& instance_columns_vec,
+           std::vector<std::vector<Poly>>&& instance_polys_vec,
+           std::vector<F>&& challenges, bool query_instance)
+      : num_circuits_(num_circuits),
+        fixed_columns_(fixed_columns),
+        fixed_polys_(fixed_polys),
+        advice_columns_vec_(std::move(advice_columns_vec)),
+        advice_blinds_vec_(std::move(advice_blinds_vec)),
+        instance_columns_vec_(std::move(instance_columns_vec)),
+        instance_polys_vec_(std::move(instance_polys_vec)),
+        challenges_(std::move(challenges)),
+        query_instance_(query_instance) {}
+
+  // Generate instance columns in evaluation form with coefficient-formed
+  // instance columns. (Batch IFFT)
+  static std::vector<std::vector<Poly>> GenerateInstancePolys(
+      Prover<PCSTy>* prover,
+      std::vector<std::vector<Evals>> instance_columns_vec,
+      bool query_instance) {
+    std::vector<std::vector<Poly>> ret;
+    ret.reserve(instance_columns_vec.size());
+
+    for (const std::vector<Evals>& instance_columns : instance_columns_vec) {
+      std::vector<Poly> instance_polys;
+      instance_polys.reserve(instance_columns.size());
+      for (const Evals& instance_column : instance_columns) {
+        Poly poly = prover->domain()->IFFT(instance_column);
+        instance_polys.push_back(std::move(poly));
+
+        if (query_instance) {
```

Comment:
After merging https://github.com/kroma-network/tachyon/pull/171,

```suggestion
        if constexpr (PCSTy::kQueryInstance) {
```